### PR TITLE
Rename old references to BlockHeaderHash and BlockHeight

### LINKED ISSUE #968: Remove remaining references to BlockHeaderHash and BlockHeight
In #905, we renamed `BlockHeaderHash` to `block::Hash` and `BlockHeight` to `block::Height`.

But there are still a few instances of the old names in comments, debug formatters, and tests. We should replace them with the new names.

### PR DESCRIPTION
**Pull request by oxarbitrage**
Closes https://github.com/ZcashFoundation/zebra/issues/968 if merged.

### CODE REVIEW

**Review by yaahc — DISMISSED** [review]
:+1:

File: book/src/dev/rfcs/0002-parallel-verification.md [hunk 1]
```diff
@@ -175,7 +175,7 @@ Verification is implemented by the following traits and services:
     data structures from a byte buffer.
 * **Semantic Verification:**
   * `ChainVerifier`: Provides a verifier service that accepts a `Block` request,
-    performs verification on the block, and responds with a `BlockHeaderHash` on
+    performs verification on the block, and responds with a `Hash` on
```
Comment (dconnolly) [inline book/src/dev/rfcs/0002-parallel-verification.md]:
```suggestion
    performs verification on the block, and responds with a `block::Hash` on
```

File: zebra-network/src/protocol/external/message.rs [hunk 2]
```diff
@@ -165,7 +165,7 @@ pub enum Message {
         /// Block locators, from newest back to genesis block.
         block_locator_hashes: Vec<block::Hash>,
 
-        /// `BlockHeaderHash` of the last desired block.
+        /// `Hash` of the last desired block.
```
Comment (dconnolly) [inline zebra-network/src/protocol/external/message.rs]:
```suggestion
        /// `block::Hash` of the last desired block.
```

File: zebra-state/src/lib.rs [hunk 3]
```diff
@@ -4,8 +4,8 @@
 //!
 //! zebra-state tracks `Blocks` using two key-value trees
 //!
-//! * BlockHeaderHash -> Block
-//! * BlockHeight -> Block
+//! * Hash -> Block
```
Comment (teor2345) [inline zebra-state/src/lib.rs]:
```suggestion
//! * block::Hash -> Block
```

**Review by teor2345 — DISMISSED** [review]
Looks good, thanks for doing this work!

We need to change every BlockHeaderHash to block::Hash, because Hash is also a standard library trait.

**Comment by oxarbitrage** [comment]
I was in doubt in regards to that but forgot to ask. Thanks for the suggestions for it, applied at https://github.com/ZcashFoundation/zebra/pull/1002/commits/f776dbf51f38915cf0d2b1fe7f3a39e163137cac

**Review by yaahc — APPROVED** [review]

**Comment by yaahc** [comment]
ty again @oxarbitrage

**Pull request merged**